src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -76,7 +76,12 @@ public void Apply(BsonClassMap classMap)
                 // if any constructors were mapped by this convention then map all the properties also
                 foreach (var property in properties)
                 {
-                    classMap.MapMember(property);
+                    var memberMap = classMap.MapMember(property);
+                    if (classMap.IsAnonymous)
+                    {
+                        var defaultValue = memberMap.DefaultValue;
+                        memberMap.SetDefaultValue(defaultValue);
```

Comment:
Calling SetDefaultValue has the side effect of allowing missing elements to be replaced with the default value during deserialization.

The BsonClassMapSerializer only does that when an *explicit* default value has been set.